tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/BlobStoreCacheModulesChooser.java

```diff
@@ -0,0 +1,83 @@
+package com.linagora.tmail.blob.blobid.list;
+
+import java.io.FileNotFoundException;
+import java.util.List;
+
+import javax.inject.Named;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.init.configuration.InjectionNames;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.MetricableBlobStore;
+import org.apache.james.blob.cassandra.cache.BlobStoreCache;
+import org.apache.james.blob.cassandra.cache.CachedBlobStore;
+import org.apache.james.blob.cassandra.cache.CassandraBlobCacheModule;
+import org.apache.james.blob.cassandra.cache.CassandraBlobStoreCache;
+import org.apache.james.blob.cassandra.cache.CassandraCacheConfiguration;
+import org.apache.james.modules.mailbox.CassandraCacheSessionModule;
+import org.apache.james.modules.mailbox.ConfigurationComponent;
+import org.apache.james.utils.PropertiesProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Names;
+
+public class BlobStoreCacheModulesChooser {
```

Comment:
Why need to copy anything related to the cache? Can't reuse the existing one?

Reply:
We can, If we edit access modifier (private -> public) in James project.
I think it has more one way, and I select one (randomly :D )

Reply:
> Why need to copy anything related to the cache? Can't reuse the existing one?

- Reason 1: the configuration object is not the same
- Reason 2: the injection logic is quite different - adapting would lead to complex design.